### ISSUE TITLE
AE non used items feature request/bug 

### ISSUE THREAD

**Issue by 0lafe** [GTNewHorizons/GT-New-Horizons-Modpack#6895]
#### Which modpack version are you using?
9.0

When making AE patterns, if you put an item that isn't consumed in the ingredients, and the outputs, when queueing multiples, it'll need 1 per recipe queued, despite using 0. Not sure how difficult it would be to change this, but it would be a nice change.

**Comment by GTNH-Afx237v7** [comment]
Not that I can code but would love some clarification on this as I don’t understand what you mean...

**Comment by 0lafe** [comment]
Say you put a circuit in a pattern that won't be used, but needs to be input. This could also apply to a flask with a fluid for example, as you'll get the empty one back. If you put that circuit or empty flask in the output of the AE, it logically should be able to know that it gets that item back and it isn't consumed. However if you queue up say 5 recipes at once, it will need 5x the circuits/empty flasks. I think a similar problem occurs with GT tools as they're similarly not lost. 

Basically even putting the non lost items into the output doesn't let AE know that it doesn't need more than 1 set

**Comment by repo-alt** [comment]
GT tools change NBT after each craft, so no way. Absolutely identical items though, well, maybe, I'll take a look.

**Comment by GTNH-Colen** [comment]
I don't understand, how does this nerf stuff?

In all seriousness though I greatly support this, it's annoying af that AE has to craft 100s of wrenches (I have 1500 odd in my AE atm) for some large recipes.

**Comment by repo-alt** [comment]
About wrenches, and why you's out of luck: Simulating all the crafts that may change the item is not really possible because crafting handler may theoretically do anything to the wrench, and AE has no way to know what is done there: it is a black box. So if the returned item is not exactly the same than that in the input, yes, it will pre-craft all the wrenches, because for all it knows, the wrenches may become unusable after 1 craft. We really should switch more to the assembler recipes after some point, to avoid the whole issue, or add some unbreakable crafting-only tools.

**Comment by S4mpsa** [comment]
The wrench problem is solved by not letting AE autocraft the tools.

**Comment by boubou19** [comment]
You still need a pool of tools in the AE if you want to mass craft something, otherwise it will be slowed down

**Comment by GTNH-Colen** [comment]
> About wrenches, and why you're out of luck: Simulating all the crafts that may change the item is not really possible because crafting handler may theoretically do anything to the wrench, and AE has no way to know what is done there: it is a black box. So if the returned item is not exactly the same than that in the input, yes, it will pre-craft all the wrenches, because for all it knows, the wrenches may become unusable after 1 craft. We really should switch more to the assembler recipes after some point, to avoid the whole issue, or add some unbreakable crafting-only tools.

Yea this was what I thought was implied before, to make special tools that won't ever break for AE recipes such as this. I'd agree that for future reference we should probs be using more assembler recipes though yea.

**Comment by boubou19** [comment]
also, allowing oredict substitutions with wrench does work perfectly

**Comment by 0lafe** [comment]
Can we still get this on exactly similar items? The tools concept was more explaining the looping concept, Im much less invested in that working

**Comment by Prometheus0000** [comment]
Maybe you could give some other examples then? Where does this actually occur? With empty cells?

**Comment by GTNH-Afx237v7** [comment]
Flawed cubic zirconia which is made in the EBF with one yttrium oxide dust and 10 zirconium dust. The zirconium dust is consumed, the yttrium oxide dust is not.

**Comment by 0lafe** [comment]
Empty cells/flasks, integrated circuits, molds/shapes, and anything else that "doesn't get consumed in the process"

**Comment by Anarack** [comment]
Blood orbs is another, though most of those are assembler recipes where the thing ends up staying.

**Comment by repo-alt** [comment]
integrated circuits, blood orbs, molds - they are all not getting into output, or are they?

**Comment by 0lafe** [comment]
They never end up in the output of the machine, but they're "done" after the recipe is ran. Unless that poses a specific issue with AE, they will return into the system and can appear as a "product" of the recipe.
